torch/distributions.py

```diff
@@ -163,6 +191,17 @@ def log_prob(self, value):
 
         return p.gather(-1, value.unsqueeze(-1)).squeeze(-1).log()
 
+    def enumerate_support(self):
+        batch_shape, event_size = self.probs.shape[:-1], self.probs.shape[-1]
+        values = torch.arange(event_size).long()
+        values = values.view((-1,) + (1,) * len(batch_shape))
+        values = values.expand((-1,) + batch_shape)
+        if self.probs.is_cuda:
+            values = values.cuda()
```

Comment:
you probably want to retain the device: `values = values.cuda(self.probs.get_device())`

Reply:
Thanks! I'll also make changes in Pyro.